pkg/pillar/scripts/device-steps.sh

```diff
@@ -164,6 +167,25 @@ if [ ! -d $PERSISTDIR/log ]; then
     mkdir $PERSISTDIR/log
 fi
 
+
+LAST_EVE_VERSION=""
+if [ -f $LAST_IMAGE_VERSION_FILE ];
+then
+     LAST_EVE_VERSION=$(jq '.Version' < $LAST_IMAGE_VERSION_FILE| tr -d '"')
+fi
+
+# Run upgradeconverter
+if [ "$LAST_EVE_VERSION" = "$EVE_VERSION" ];
+then
+     echo "$(date -Ins -u) device-steps: Same EveVersion as Last. $EVE_VERSION"
+     echo "$(date -Ins -u) device-steps: Not Running upgradeconverter"
+else
+    echo "$(date -Ins -u) device-steps: LAST_EVE_VERSION($LAST_EVE_VERSION) != EVE_VERSION($EVE_VERSION)"
+    echo "$(date -Ins -u) device-steps: Starting upgradeconverter"
+    $BINDIR/upgradeconverter -a "convert"
```

Comment:
Why is the "convert" argument passed to it here? What's the point of passing an argument?

Reply:
As I said, the upgrade converter handles 2 things:

1) Convert Old Config to New Config at the beginning of bootup
2) Clean up old config.

The idea is to isolate rest of the pillar code from Version knowledge. Only the Upgrade Converter will deal with any necessary differences between the 2 versions and convert / generate any required by the new version. 

Hence passing "convert" as an argument..